adapters/mysql2/lib/opentelemetry/adapters/mysql2/patches/client.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Mysql2
+      module Patches
+        # Module to prepend to Mysql2::Client for instrumentation
+        module Client
+          def query(sql, options = {})
+            response = nil
+
+            tracer.in_span(
+              database_span_name,
+              attributes: client_attributes.merge(
+                'db.statement' => sql
+              ),
+              kind: :client
+            ) do
+              response = super(sql, options)
+            end
+
+            response
```

Comment:
The local `response` doesn’t really buy us anything here. `tracer.in_span(...) do ... end` will return the result of the block.

Reply:
👍 makes sense, updated